core/shared/src/main/scala/monocle/function/Index.scala

```diff
@@ -45,6 +46,13 @@ object Index extends IndexFunctions {
   /** Std instances */
   /** *********************************************************************************************
     */
+  implicit def seqIndex[A, S[B] <: SeqOps[B, S, S[B]]]: Index[S[A], Int, A] =
+    Index(i =>
+      if (i < 0) Optional.void
+      else
+        Optional[S[A], A](s => Some(i).filter(s.indices.contains).map(s.apply))(a => s => Try(s.updated(i, a)).getOrElse(s))
```

Comment:
is it faster to do `Some(i).filter(s.indices.contains)` than to catch the exception if the indice doesn't exist?

Reply:
Good question. It depends...
After thinking about it, I realized that `Try(s(i)).toOption` is the best default implementation for sequences as its only drawback in terms of performance is the exception handling overhead.
On the other hand, implementations of `IndexedSeqOps` are supposed to have efficient `indices` method, so I believe that it makes sense to create a corresponding `Index` in the future.